### PR TITLE
Annotate more Ledger API tests with authorisation test evidence.

### PR DESCRIPTION
**Pull request by skisel-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/AdminServiceCallAuthTests.scala [hunk 1]
```diff
@@ -17,54 +17,87 @@ trait AdminServiceCallAuthTests extends SecuredServiceCallAuthTests {
...
   }
-  it should "deny calls with an expired admin token" in {
+  it should "deny calls with an expired admin token" taggedAs securityAsset.setAttack(
+    attack(threat = "Exploit an expired admin token")
+  ) in {
     expectUnauthenticated(serviceCallWithToken(canReadAsAdminExpired))
   }
-  it should "deny calls with a read-only token" in {
+  it should "deny calls with a read-only token" taggedAs securityAsset.setAttack(
+    attack(threat = "Exploit a read-only token")
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/AdminServiceCallAuthTests.scala:31]:
```suggestion
    attack(threat = "Exploit a read-only token for a random party")
```

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/AdminServiceCallAuthTests.scala [hunk 2]
```diff
@@ -17,54 +17,87 @@ trait AdminServiceCallAuthTests extends SecuredServiceCallAuthTests {
...
+  ) in {
     expectPermissionDenied(serviceCallWithToken(canReadAsRandomParty))
   }
-  it should "deny calls with a read/write token" in {
+  it should "deny calls with a read/write token" taggedAs securityAsset.setAttack(
+    attack(threat = "Exploit read/write token")
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/AdminServiceCallAuthTests.scala:36]:
```suggestion
    attack(threat = "Exploit read/write token for a random party")
```

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala [hunk 3]
```diff
@@ -53,31 +54,46 @@ trait MultiPartyServiceCallAuthTests extends SecuredServiceCallAuthTests {
   // - ActAs parties from the request require ActAs claims in the token
   // - ReadAs parties from the request require ReadAs or ActAs claims in the token
 
-  it should "allow single-party calls authorized to exactly the submitter (party)" in {
+  it should "allow single-party calls authorized to exactly the submitter (party)" taggedAs securityAsset
+    .setHappyCase(
+      "Single Party Ledger API client can make a call to exactly the submitter (party)"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala:59]:
```suggestion
      "Ledger API client can make a single-party calls authorized to exactly the submitter (party)"
```

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala [hunk 4]
```diff
@@ -53,31 +54,46 @@ trait MultiPartyServiceCallAuthTests extends SecuredServiceCallAuthTests {
...
         TokenParties(List(singleParty), List.empty),
         RequestSubmitters(singleParty, List.empty, List.empty),
       )
     )
   }
-  it should "allow single-party calls authorized to exactly the submitter (actAs)" in {
+
+  it should "allow single-party calls authorized to exactly the submitter (actAs)" taggedAs securityAsset
+    .setHappyCase(
+      "Single Party Ledger API client can make a call to exactly the submitter (actAs)"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala:71]: As above

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala [hunk 5]
```diff
@@ -53,31 +54,46 @@ trait MultiPartyServiceCallAuthTests extends SecuredServiceCallAuthTests {
...
         TokenParties(List(singleParty), List.empty),
         RequestSubmitters("", List(singleParty), List.empty),
       )
     )
   }
-  it should "allow single-party calls authorized to exactly the submitter (party, actAs, and readAs)" in {
+
+  it should "allow single-party calls authorized to exactly the submitter (party, actAs, and readAs)" taggedAs securityAsset
+    .setHappyCase(
+      "Single Party Ledger API client can make a call to exactly the submitter (party, actAs, and readAs)"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala:83]: As above

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala [hunk 6]
```diff
@@ -53,31 +54,46 @@ trait MultiPartyServiceCallAuthTests extends SecuredServiceCallAuthTests {
...
         TokenParties(List(singleParty), List.empty),
         RequestSubmitters(singleParty, List(singleParty), List(singleParty)),
       )
     )
   }
-  it should "allow single-party calls authorized to a superset of the required parties (party)" in {
+
+  it should "allow single-party calls authorized to a superset of the required parties (party)" taggedAs securityAsset
+    .setHappyCase(
+      "Single Party Ledger API client can make a call to a superset of the required parties (party)"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala:95]: As above

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala [hunk 7]
```diff
@@ -86,47 +102,70 @@ trait MultiPartyServiceCallAuthTests extends SecuredServiceCallAuthTests {
...
         TokenParties(List.empty, List(singleParty)),
         RequestSubmitters(singleParty, List.empty, List.empty),
       )
     )
   }
-  it should "deny single-party calls authorized in read-only mode (actAs)" in {
+
+  it should "deny single-party calls authorized in read-only mode (actAs)" taggedAs securityAsset
+    .setAttack(
+      attack(threat = "Exploit a single-party calls authorize in read-only mode (actAs)")
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala:143]:
```suggestion
      attack(threat = "Exploit a single-party calls authorized in read-only mode (actAs)")
```

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala [hunk 8]
```diff
@@ -135,39 +174,58 @@ trait MultiPartyServiceCallAuthTests extends SecuredServiceCallAuthTests {
     )
   }
 
-  it should "allow multi-party calls authorized to exactly the required parties" in {
+  it should "allow multi-party calls authorized to exactly the required parties" taggedAs securityAsset
+    .setHappyCase(
+      "Multi-party Ledger API client can make a call to exactly the required parties"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala:179]:
```suggestion
      "Ledger API client can make a multi-party call to exactly the required parties"
```

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala [hunk 9]
```diff
@@ -135,39 +174,58 @@ trait MultiPartyServiceCallAuthTests extends SecuredServiceCallAuthTests {
...
         TokenParties(actAs, readAs),
         RequestSubmitters("", actAs, readAs),
       )
     )
   }
-  it should "allow multi-party calls authorized to a superset of the required parties" in {
+
+  it should "allow multi-party calls authorized to a superset of the required parties" taggedAs securityAsset
+    .setHappyCase(
+      "Multi-party Ledger API client can make a call to a superset of the required parties"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala:191]: As above

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala [hunk 10]
```diff
@@ -135,39 +174,58 @@ trait MultiPartyServiceCallAuthTests extends SecuredServiceCallAuthTests {
...
         TokenParties(randomActAs ++ actAs, randomReadAs ++ readAs),
         RequestSubmitters("", actAs, readAs),
       )
     )
   }
-  it should "allow multi-party calls with all parties authorized in read-write mode" in {
+
+  it should "allow multi-party calls with all parties authorized in read-write mode" taggedAs securityAsset
+    .setHappyCase(
+      "Multi-party Ledger API client can make a call with all parties authorized in read-write mode"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala:203]: As above

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala [hunk 11]
```diff
@@ -135,39 +174,58 @@ trait MultiPartyServiceCallAuthTests extends SecuredServiceCallAuthTests {
...
         TokenParties(actAs ++ readAs, List.empty),
         RequestSubmitters("", actAs, readAs),
       )
     )
   }
-  it should "allow multi-party calls with actAs parties spread across party and actAs fields" in {
+
+  it should "allow multi-party calls with actAs parties spread across party and actAs fields" taggedAs securityAsset
+    .setHappyCase(
+      "Multi-party Ledger API client can make a call with actAs parties spread across party and actAs fields"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala:215]: As above

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala [hunk 12]
```diff
@@ -135,39 +174,58 @@ trait MultiPartyServiceCallAuthTests extends SecuredServiceCallAuthTests {
...
         TokenParties(actAs, readAs),
         RequestSubmitters(actAs.head, actAs.tail, readAs),
       )
     )
   }
-  it should "allow multi-party calls with actAs parties duplicated in the readAs field" in {
+
+  it should "allow multi-party calls with actAs parties duplicated in the readAs field" taggedAs securityAsset
+    .setHappyCase(
+      "Multi-party Ledger API client can make a call with actAs parties duplicated in the readAs field"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/MultiPartyServiceCallAuthTests.scala:227]: As above

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala [hunk 13]
```diff
@@ -25,16 +26,27 @@ trait ReadOnlyServiceCallAuthTests extends ServiceCallWithMainActorAuthTests {
...
-  it should "deny calls with an expired read-only token" in {
+  it should "deny calls with an expired read-only token" taggedAs securityAsset.setAttack(
+    attack(threat = "Exploit an expired read-only token")
+  ) in {
     expectUnauthenticated(serviceCallWithToken(canReadAsMainActorExpired))
   }
-  it should "allow calls with explicitly non-expired read-only token" in {
+  it should "allow calls with explicitly non-expired read-only token" taggedAs securityAsset
+    .setHappyCase(
+      "Ledger API client can connect with explicitly non-expired read-only token"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala:36]:
```suggestion
      "Ledger API client can make a call with explicitly non-expired read-only token"
```

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala [hunk 14]
```diff
@@ -25,16 +26,27 @@ trait ReadOnlyServiceCallAuthTests extends ServiceCallWithMainActorAuthTests {
...
+    ) in {
     successfulBehavior(serviceCallWithToken(canReadAsMainActorExpiresTomorrow))
   }
-  it should "allow calls with read-only token without expiration" in {
+  it should "allow calls with read-only token without expiration" taggedAs securityAsset
+    .setHappyCase(
+      "Ledger API client can connect with read-only token without expiration"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala:42]:
```suggestion
      "Ledger API client can make a call with read-only token without expiration"
```

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala [hunk 15]
```diff
@@ -25,16 +26,27 @@ trait ReadOnlyServiceCallAuthTests extends ServiceCallWithMainActorAuthTests {
...
+    ) in {
     successfulBehavior(serviceCallWithToken(canReadAsMainActor))
   }
-  it should "allow calls with user token that can-read-as main actor" in {
+  it should "allow calls with user token that can-read-as main actor" taggedAs securityAsset
+    .setHappyCase(
+      "Ledger API client can connect with read-only token without expiration"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala:48]:
```suggestion
      "Ledger API client can make a call with user token that can-read-as main actor"
```

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala [hunk 16]
```diff
@@ -46,26 +58,45 @@ trait ReadOnlyServiceCallAuthTests extends ServiceCallWithMainActorAuthTests {
...
-  it should "deny calls with an expired read/write token" in {
+  it should "deny calls with an expired read/write token" taggedAs securityAsset.setAttack(
+    attack(threat = "Exploit an expired read/write token")
+  ) in {
     expectUnauthenticated(serviceCallWithToken(canActAsMainActorExpired))
   }
-  it should "allow calls with explicitly non-expired read/write token" in {
+  it should "allow calls with explicitly non-expired read/write token" taggedAs securityAsset
+    .setHappyCase(
+      "Ledger API client can connect with non-expired read/write token"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala:86]:
```suggestion
      "Ledger API client can make a call with non-expired read/write token"
```

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala [hunk 17]
```diff
@@ -46,26 +58,45 @@ trait ReadOnlyServiceCallAuthTests extends ServiceCallWithMainActorAuthTests {
...
+    ) in {
     successfulBehavior(serviceCallWithToken(canActAsMainActorExpiresTomorrow))
   }
-  it should "allow calls with read/write token without expiration" in {
+  it should "allow calls with read/write token without expiration" taggedAs securityAsset
+    .setHappyCase(
+      "Ledger API client can connect with read/write token without expiration"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala:92]:
```suggestion
      "Ledger API client can make a call with read/write token without expiration"
```

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala [hunk 18]
```diff
@@ -46,26 +58,45 @@ trait ReadOnlyServiceCallAuthTests extends ServiceCallWithMainActorAuthTests {
...
+    ) in {
     successfulBehavior(serviceCallWithToken(canActAsMainActor))
   }
-  it should "allow calls with user token that can-act-as main actor" in {
+  it should "allow calls with user token that can-act-as main actor" taggedAs securityAsset
+    .setHappyCase(
+      "Ledger API client can connect with user token that can-act-as main actor"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala:98]:
```suggestion
      "Ledger API client can make a call with user token that can-act-as main actor"
```

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala [hunk 19]
```diff
@@ -74,16 +105,24 @@ trait ReadOnlyServiceCallAuthTests extends ServiceCallWithMainActorAuthTests {
     )
   }
 
-  it should "allow calls with the correct ledger ID" in {
+  it should "allow calls with the correct ledger ID" taggedAs securityAsset.setHappyCase(
+    "Ledger API client can connect with the correct ledger ID"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala:109]:
```suggestion
    "Ledger API client can make a call with the correct ledger ID"
```

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala [hunk 20]
```diff
@@ -74,16 +105,24 @@ trait ReadOnlyServiceCallAuthTests extends ServiceCallWithMainActorAuthTests {
...
   }
-  it should "deny calls with a random ledger ID" in {
+  it should "deny calls with a random ledger ID" taggedAs securityAsset.setAttack(
+    attack(threat = "Exploit a random ledger ID")
+  ) in {
     expectPermissionDenied(serviceCallWithToken(canReadAsMainActorRandomLedgerId))
   }
-  it should "allow calls with the correct participant ID" in {
+  it should "allow calls with the correct participant ID" taggedAs securityAsset.setHappyCase(
+    "Ledger API client can connect with the correct participant ID"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadOnlyServiceCallAuthTests.scala:119]:
```suggestion
    "Ledger API client can makes a call with the correct participant ID"
```

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadWriteServiceCallAuthTests.scala [hunk 21]
```diff
@@ -16,63 +17,107 @@ trait ReadWriteServiceCallAuthTests extends ServiceCallWithMainActorAuthTests {
...
-  it should "deny calls with an expired read/write token" in {
+  it should "deny calls with an expired read/write token" taggedAs securityAsset.setAttack(
+    attack(threat = "Exploit an expired read/write token")
+  ) in {
     expectUnauthenticated(serviceCallWithToken(canActAsMainActorExpired))
   }
-  it should "allow calls with explicitly non-expired read/write token" in {
+  it should "allow calls with explicitly non-expired read/write token" taggedAs securityAsset
+    .setHappyCase(
+      "Ledger API client can connect with explicitly non-expired read/write token"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadWriteServiceCallAuthTests.scala:27]:
```suggestion
      "Ledger API client can make a call with explicitly non-expired read/write token"
```
Similar changes across this file

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadWriteServiceCallAuthTests.scala [hunk 22]
```diff
@@ -16,63 +17,107 @@ trait ReadWriteServiceCallAuthTests extends ServiceCallWithMainActorAuthTests {
...
+  it should "deny calls with read-only token without expiration" taggedAs securityAsset.setAttack(
+    attack(threat = "Exploit a read-only token without expiration")
+  ) in {
     expectPermissionDenied(serviceCallWithToken(canReadAsMainActor))
   }
-
-  it should "allow calls with the correct ledger ID" in {
+  it should "allow calls with the correct ledger ID" taggedAs securityAsset
+    .setHappyCase(
+      "Ledger API client can connect with "
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadWriteServiceCallAuthTests.scala:80]: Something is missing?

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadWriteServiceCallAuthTests.scala [hunk 23]
```diff
@@ -16,63 +17,107 @@ trait ReadWriteServiceCallAuthTests extends ServiceCallWithMainActorAuthTests {
...
-  it should "allow calls with the correct participant ID" in {
+  it should "allow calls with the correct participant ID" taggedAs securityAsset
+    .setHappyCase(
+      "Ledger API client can connect with the correct participant ID"
+    ) in {
     expectSuccess(serviceCallWithToken(canActAsMainActorActualParticipantId))
   }
-  it should "deny calls with a random participant ID" in {
+  it should "deny calls with a random participant ID" taggedAs securityAsset.setAttack(
+    attack(threat = "Exploit a")
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadWriteServiceCallAuthTests.scala:96]: Missing text?

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadWriteServiceCallAuthTests.scala [hunk 24]
```diff
@@ -16,63 +17,107 @@ trait ReadWriteServiceCallAuthTests extends ServiceCallWithMainActorAuthTests {
...
     expectSuccess(serviceCallWithToken(canActAsMainActorActualApplicationId))
   }
-  it should "deny calls with a random application ID" in {
+  it should "deny calls with a random application ID" taggedAs securityAsset.setAttack(
+    attack(threat = "Exploit a random application ID")
+  ) in {
     expectPermissionDenied(serviceCallWithToken(canActAsMainActorRandomApplicationId))
   }
-  it should "allow calls with an empty application ID for a token with an application id" in {
+  it should "allow calls with an empty application ID for a token with an application id" taggedAs securityAsset
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadWriteServiceCallAuthTests.scala:111]: Same old problem can be found here as was the case in the [CompletionStreamAuthIT.scala](https://github.com/digital-asset/daml/pull/13445/files#diff-4b91578158d26be2da1aad87e8140d429bb48dfb8150d6f18b9c89e0236b17af)

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/UnsecuredServiceCallAuthTests.scala [hunk 25]
```diff
@@ -2,11 +2,14 @@
...
 
 package com.daml.platform.sandbox.auth
+import com.daml.test.evidence.scalatest.ScalaTestSupport.Implicits._
 
 trait UnsecuredServiceCallAuthTests extends ServiceCallAuthTests {
   behavior of serviceCallName
 
-  it should "allow unauthenticated calls" in {
+  it should "allow unauthenticated calls" taggedAs securityAsset.setHappyCase(
+    "Ledger API client can connect without authentication"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/UnsecuredServiceCallAuthTests.scala:11]:
```suggestion
    "Ledger API client can make a call without authentication"
```

File: ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadWriteServiceCallAuthTests.scala (hunk 22 above)
Comment (skisel-da) [reply ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/ReadWriteServiceCallAuthTests.scala:80]: oops, will fix it

File: ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/CompletionStreamAuthIT.scala [hunk 26]
```diff
@@ -51,19 +52,30 @@ final class CompletionStreamAuthIT
...
+  it should "deny calls with a random application ID" taggedAs securityAsset.setAttack(
+    attack(threat = "Exploit a call with a random application ID")
+  ) in {
     expectPermissionDenied(serviceCallWithToken(canReadAsMainActorRandomApplicationId))
   }
 
-  it should "allow calls with an empty application ID for a token with an application id" in {
+  it should "allow calls with an application ID present in the message and a token with an empty application ID" taggedAs securityAsset
+    .setHappyCase(
+      "Ledger API client can make a call with an empty application ID for a token with an application id"
```
Comment (mziolekda) [inline ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/CompletionStreamAuthIT.scala:70]: Can you make it consistent with the case text? Same applies to other places where you have rephrased the case text

**Review by mziolekda — APPROVED** [review]

**Pull request merged**